### ISSUE TITLE
PackageRevision tasks continuously grows in size

### ISSUE THREAD

**Issue by ChristopherFry** [kptdev/kpt#3525]
With the PackageRevision tasks list (in `spec.tasks`) always being appended to anytime a package revision is updated, we'll need a way to compact this list after a number of tasks occur to ensure the list does not continuously grow in size.

**Comment by johnbelamaric** [comment]
I would probably classify an eternally growing list as a bug rather than an enhancement. Eventually this will reach various limits and cause different parts of the pipeline to puke.

**Comment by ChristopherFry** [comment]
> I would probably classify an eternally growing list as a bug rather than an enhancement. Eventually this will reach various limits and cause different parts of the pipeline to puke.

Agree. Updated the title of this issue, removed the enhancement label, and added the bug label.

**Comment by liamfallon** [comment]
Duplicated to Porch in the Nephio project: https://github.com/nephio-project/nephio/issues/635
